html/semantics/embedded-content/media-elements/playing-the-media-resource/loop.html

```diff
@@ -0,0 +1,57 @@
+<!doctype html>
+<title>loop</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/media.js"></script>
+<div id="log"></div>
+<script>
+// Test the interaction of the seek algorithm's "Set the current playback
+// position to the given new playback position" with the loop step in "When the
+// current playback position reaches the end of the media resource when the
+// direction of playback is forwards". When the seek algorithm sets the current
+// playback position, that triggers the loop step which in turn triggers a new
+// seek. That seek aborts the first seek, which means that its timeupdate and
+// seeked events are never fired.
+function await_seeked(a, callback) {
+  var events = [];
+  a.onseeking = a.ontimeupdate = a.onseeked = function(e) {
+    events.push(e.type);
+    if (e.type == 'seeked') {
+      callback(events);
+    }
+  };
+}
+
+async_test(function(t) {
+  var a = new Audio(getAudioURI('/media/sound_5'));
+  a.loop = true;
+  a.onloadedmetadata = t.step_func(function() {
+    a.currentTime = a.duration;
+    await_seeked(a, t.step_func(function(events) {
+      assert_equals(a.currentTime, 0);
+      assert_array_equals(events, ['seeking', 'timeupdate', 'seeked']);
```

Comment:
Shouldn't there be two `seeking` events? First, from setting `a.currentTime = a.duration`

https://html.spec.whatwg.org/multipage/media.html#dom-media-currenttime

"otherwise, it must set the official playback position to the new value and then seek to the new value."

Step 10 queues the `seeking` event.

Step 11 changes current playback position.

Then, https://html.spec.whatwg.org/multipage/media.html#reaches-the-end (which I believe is invoked regardless of "potentially playing", right?) step 1 seeks again, which aborts the first seek, but doesn't remove the queued event from the event queue. So step 10 will queue another `seeking` event, and then 16 and 17 queue `timeupdate` and `seeked`.

Reply:
Your interpretation of the spec looks correct to me, but I think the interop issue was precisely around whether "reaches the end" is triggered when seeking and not just when playing. See the note:
> The word "reaches" here does not imply that the current playback position needs to have changed during normal playback; it could be via seeking, for instance.

In 2017 I pushed https://github.com/web-platform-tests/wpt/pull/789/commits/a525f4775eac1f5ffdaf7571aecf0ff1e28b4aa5 to to expect a single `seeking` event instead of two, saying "update to presumably not match spec, but maybe pass".

Reverting back to the previous state, Chrome and Firefox both fail. But there's something weird with Safari that isn't just about a user gesture, so I'll need to dig into that.